### PR TITLE
Add ADRs for `ConsensusChannel` ledger implementation, Peer Messaging, & Engine Crank Scheduling

### PR DESCRIPTION
**Pull request by NiloCK**
PR adds three ADRs describing components of the ledger protocol as they currently exist, with:

`0003-consensus-ledger-channels.md`
- background on the motivating problem
- brief overview of the `async` vs `consensus` update options
- medium-resolution description of `ConsensusChannel` data structures, API
- brief descriptions of how the `ConsensusChannel` is handled by the various protocols

`0004-proposal-messaging.md`
- descriptions of
  - Redundant delivery of unacknowledged messages
  - guaranteed-delivery message service
  - specific requests for missing proposals
- description of caveats on current redundancy scheme

`0005-proposal-processing.md`
- describes the engine's mechanism for "kicking" protocols that previously stalled out due to a ledger channel not being ready

Limitations:
- `0005` is a stub, really. I do not have a great understanding of other mechanisms that were on the table. Someone could pick it up if they wanted to expand with some `Considered Options`.
- I did not break `0003` into package readmes, which was suggested (reasonably!). I'm not convinced there's much value in readmes outside of the the root `go-nitro` and for `package client` (the consumption endpoint). Specific package & implementation documentation should live on the godoc of the exported API and the package godoc itself.

---

- [x] I have assigned myself to this PR
- [x] I have linked the appropriate github issue
- [x] I have assigned this PR to the appropriate GitHub project
- [x] I have assigned this PR to the appropriate GitHub Milestone

### CODE REVIEW

File: .adr/0003-consensus-ledger-channels.md
Comment (geoknee) [inline .adr/0003-consensus-ledger-channels.md:111]: sp

**Review by geoknee — COMMENTED** [review]
This is looking great so far. I want to give it a closer read before approving, but here is some rapid feedback. 

It would be even better to include a little material on:
1. alternative ways we could protect against messages being dropped (e.g. a jumpstart mechanism, or a message service guaranteeing delivery)

2. alternative ways we could protect against messages being reordered (crank until idle approach, with the advantage of not sending redundant info but disadvantage of empty cranks)

3. A little on the "governance handoff" between channels and consensus channels. You can steal what you like from here https://www.notion.so/statechannels/Channels-ConsensusChannels-Objectives-9701e24e2bc1491a83fae375e4e4c64a#9c41424d07694150bdc6cf373309757a

**Comment by NiloCK** [comment]
> It would be even better to include a little material on:

Good suggestions.

**Comment by geoknee** [comment]
I think we ought to also describe the necessity of having something outside the protocol (namely, the engine) process things in the right order (i.e. crank objectives in the right order, by looking at a proposals objective id and turn number). https://github.com/statechannels/go-nitro/pull/582

**Review by geoknee — APPROVED** [review]
Approving now, but would be keen for @lalexgap to cast his eye over it before its merged (as well as other comments addressed if possible).

File: .adr/0003-consensus-ledger-channels.md
Comment (lalexgap) [inline .adr/0003-consensus-ledger-channels.md:44]: It might be nice to just embed some of the content of the notion doc in this ADR so people can understand what async virtual funding is without having to access a private(I think?) notion doc.

File: .adr/0003-consensus-ledger-channels.md
Comment (lalexgap) [inline .adr/0003-consensus-ledger-channels.md:73]:
```suggestion
The consensus update ledger channel implementation exists as the `struct ConsensusChannel` export from `package consensus_channel`.
```

File: .adr/0003-consensus-ledger-channels.md
Comment (lalexgap) [inline .adr/0003-consensus-ledger-channels.md:125]:
```suggestion
Clients include all unacknowledged proposals with each message. In the context of a healthy network, this might be expected to result in occasional double transmissions and rare triple transmissions. This expectation is subject to future benchmarking / simulation (TODO).
```

File: .adr/0003-consensus-ledger-channels.md
Comment (lalexgap) [inline .adr/0003-consensus-ledger-channels.md:119]: It might be worth splitting this out into its own ADR?

**Pull request merged**
